### DISCUSSION TITLE
brew on macos and choco on windows support

### DISCUSSION THREAD

**Discussion by mhmdio** [Noovolari/leapp#51] [General] [answered]
Feature request:

https://brew.sh/ to install the app on mac.
https://chocolatey.org/ to install the app on windows.

**Comment by andreacavagna01** [comment]
Great idea! 
Have you ever posted some app on Brew or Chocolatey?

**Reply by gep13** [reply]
In terms of creating a Chocolatey package, we have some docs here: https://docs.chocolatey.org/en-us/create/ that will hopefully help.  If you install Chocolatey and then run `choco new leapp` it will scaffold you out a new package, with some suggestions in the generated files about how to proceed.  You will also see a number of existing packages in this [repository](https://github.com/chocolatey-community/chocolatey-coreteampackages) so if you are looking for some help on a specific Chocolatey helper function, I would search in that repository to see example uses of it.

**Reply by andreacavagna01** [reply]
Thanks, I will look at it!

**Reply by mhmdio** [reply]
No, but I can help with this at the weekends if you want.

**Reply by andreacavagna01** [reply]
absolutely! let's join our discord chat and meet the team :)

https://discord.gg/Pt5V2Sz6

**Reply by mhmdio** [reply]
The link has expired, can you send it again?
Thanks.

**Reply by andreacavagna01** [reply]
https://discord.gg/r8vG4c6

I'm sorry 👍

**Reply by sheldonhull** [reply]
Big fan of chocolatey, but wanted to mention that Scoop might be easier initially as it's in a centralized repo. 
I know Go based tooling has automatic Scoop package generation, so maybe using Scoop will get this up and running much more easily. I've got a lot of choco experience but it requires auto package updating and has some nuances to handling that those not familiar with it specifically might find a bit more confusing than the more "bash/shell" oriented approach of Scoop. 

https://github.com/lukesampson/scoop/wiki/App-Manifests

**Reply by andreacavagna01** [reply]
I Will look at Scoop, never used it, usually, I use brew, to me, have this packager support comes with the version with Leapp client and [Daemon](https://github.com/Noovolari/leapp-daemon). 
@ericvilla and @pethron I think Scoop is something you should watch

**Comment by ericvilla** [comment]
We’ve to understand how to fit it in both Desktop App and Daemon installation process.
It would be a nice thing if we’re able to create a single installation workflow with Scoop.
I’ll take some time to test it on both the Desktop App and the Daemon!

**Reply by ericvilla** [reply]
@sheldonhull, based on your availability, we could keep in touch, so you can introduce me to the tool! Then we can study together how it can fit with Leapp

**Reply by sheldonhull** [reply]
Would love too. On macOS but if I can help I'll do my best for sure!

**Comment by andreacavagna01** [comment]
Can anyone help me with Chocolatey? It's been pointed out again by @robcube, @mhmdio. How should I help to complete this task?

**Reply by sheldonhull** [reply]
Did you look at scoop instead? Go releaser makes this automatic I think

**Reply by sheldonhull** [reply]
I'd be willing to do my best to help out.
I forked both repos, but realize there is something a bit challenging in this.

Normally a single repo would contain the project so I could configure `goreleaser` to build.
This is multi-repo but designed to be distributed as a single package. 
Not sure how to properly handle that. 

I have a lot of [ci config files](https://github.com/sheldonhull/ci-configuration-files) for running golanglint-ci, markdown formatting autofix, security scans, builds and more. Nothing yet for multi-repo builds.

Btw, last quick tip. Try replacing your Makefile with [go-task](https://github.com/go-task/task). It's cross-platform and works on windows. Pretty easy to copy and paste your make file into. (install direction snippet [here](https://github.com/sheldonhull/ci-configuration-files/blob/main/snippets-docs/readme-snippet-install-go-task.md) I'm personally trying to use goyek which is a go based task tool, but it's more complex.

**Reply by ericvilla** [reply]
I didn't investigate golang release process yet, but my initial idea was to configure a service around the goland-based daemon. I've already tried [kardianos](https://github.com/kardianos/service) package but it seems to be a little bit immature when it has to deal with the macOS service ecosystem. Do you know that library?

What about installing and starting the golang-based daemon service during the electron-based client installation?
IMO, it is a good way to handle the lifecycle of the daemon even if the client app fails. Moreover, it could be a game-changer when developing the CLI.

Currently, the makefile is used only to generate go-swagger documentation, so we can move go-task if it covers what is already defined in the makefile. I personally have to study both go-task and go-releaser to understand how they fit with the creation of a single bundle.

**Comment by andreacavagna01** [comment, marked as answer]
Looks like it has been added from this issue:
https://github.com/Noovolari/leapp/issues/179
